### ISSUE TITLE
[Windows][DX12] mandlebrot example crashes

### ISSUE THREAD

**Issue by Jiloc** [rdeioris/compushady#2]
After some seconds of execution, the examples crashes with the following log:

```Using device NVIDIA GeForce RTX 2080 Ti
Traceback (most recent call last):
  File ".\test_compushady.py", line 74, in <module>
    swapchain.present(target)
  File "C:\Users\franc\AppData\Local\Programs\Python\Python38\lib\site-packages\compushady\__init__.py", line 202, in present
    self.handle.present(resource.handle, x, y)
Exception: unable to Present() Swapchain: Istanza del dispositivo GPU sospesa. Utilizzare GetDeviceRemovedReason per determinare l'azione appropriata.

ID3D12CommandQueue::ExecuteCommandLists: Command lists must be successfully closed before execution.
ID3D12Device::RemoveDevice: Device removal has been triggered for the following reason (DXGI_ERROR_INVALID_CALL: There is strong evidence that the application has performed an illegal or undefined operation, and such a condition could not be returned to the application cleanly through a return code).
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12CommandList::Dispatch: No pipeline state has been set in this command list.  The runtime will use a default no-op pipeline state.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12CommandQueue::ExecuteCommandLists: Command lists must be successfully closed before execution.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12GraphicsCommandList::*: This API cannot be called on a closed command list.
ID3D12CommandQueue::ExecuteCommandLists: Command lists must be successfully closed before execution.
```

**Comment by HenBOMB** [comment]
It's a problem with `Buffer.upload()`, it's eating up all of your RAM and the program is just crashing.

**Comment by rdeioris** [comment]
@Jiloc @HenBOMB should be fixed in 0.17.2
